### PR TITLE
feat: #RDV-16 take appointment + alert

### PR DESCRIPTION
**Pull request by niko-amosse**
## Describe your changes

- link back to front to take an appointemnt
- display alerts when we take an appointment

## Issue ticket number and link

[RDV-16](https://jira.support-ent.fr/browse/RDV-16)

## Checklist before requesting a review (magic string, indentation, comment/documentation...)
- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

**Comment by qperie** [comment]
Respecte bien le format de nom de commit pour le titre de ta PR (manque notamment le # ici en l'occurrence)

File: backend/src/main/java/fr/openent/appointments/controller/AppointmentController.java [hunk 1]
```diff
@@ -82,7 +104,7 @@ public void createAppointment(final HttpServerRequest request) {
             .onFailure(err -> {
                 String errorMessage = "Failed to create appointment";
                 LogHelper.logError(this, "createAppointment", errorMessage, err.getMessage());
-                renderError(request, new JsonObject().put("error", errorMessage));
+                if(!request.isEnded()) renderError(request, new JsonObject().put("error", errorMessage));
```
Comment (qperie) [inline backend/src/main/java/fr/openent/appointments/controller/AppointmentController.java:107]: petit magic string ici

File: backend/src/main/java/fr/openent/appointments/repository/impl/DefaultGridRepository.java [hunk 2]
```diff
@@ -232,8 +241,26 @@ public Future<JsonObject> closeAllPassedGrids() {
     private Future<Optional<Grid>> insert(GridPayload grid, String userId) {
         Promise<Optional<Grid>> promise = Promise.promise();
 
-        List<String> sqlColumns = Arrays.asList(NAME, OWNER_ID, STRUCTURE_ID, BEGIN_DATE, END_DATE,  CREATION_DATE, UPDATING_DATE,
-                COLOR, DURATION, PERIODICITY, TARGET_PUBLIC_LIST_ID, VISIO_LINK, PLACE, DOCUMENT_ID, PUBLIC_COMMENT, STATE);
+        String frenchNow = ZonedDateTime.now(ZoneId.of("Europe/Paris"))
+                .format(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss"));
```
Comment (qperie) [inline backend/src/main/java/fr/openent/appointments/repository/impl/DefaultGridRepository.java:245]: magic strings

File: backend/src/main/java/fr/openent/appointments/service/AppointmentService.java [hunk 3]
```diff
@@ -6,11 +6,30 @@
...
+     * @param timeSlotId The time slot id
+     * @return if the time slot is available
+     */
+    public Future<Boolean> checkIfTimeSlotIsAvailable(Long timeSlotId);
+
     /**
      *  Create an appointment linked to a time slot
      *
      *  @param timeSlotId The time slot id
+     *  @param userId The user id
```
Comment (qperie) [inline backend/src/main/java/fr/openent/appointments/service/AppointmentService.java:31]: tant qu'à faire maj la doc tu peux ajouter le param isVisio

**Review by Mattox0 — APPROVED** [review]

File: backend/src/main/java/fr/openent/appointments/core/constants/Constants.java [hunk 4]
```diff
@@ -36,6 +37,7 @@ public class Constants {
     public static final String APPOINTMENTS_ADDRESS = "fr.openent.appointments";
     public static final String BODY = "body";
     public static final String FUNCTIONS = "functions";
+    public static final String FRENCH_NOW = "NOW() AT TIME ZONE 'Europe/Paris'";
```
Comment (alicedraillard) [inline backend/src/main/java/fr/openent/appointments/core/constants/Constants.java:40]: - sépare "NOW() AT TIME ZONE" et  'Europe/Paris'
- vérifie si tu as besoin de mettre des \ avant les ' pour 'Europe/Paris'

File: backend/src/main/java/fr/openent/appointments/service/impl/DefaultTimeSlotService.java [hunk 5]
```diff
@@ -117,6 +117,30 @@ public Future<Boolean> checkIfTimeSlotIsAvailable(Long timeSlotId) {
         return promise.future();
     }
 
+    @Override
+    public Future<Boolean> checkIfTimeSlotIsVisio(Long timeSlotId) {
```
Comment (alicedraillard) [inline backend/src/main/java/fr/openent/appointments/service/impl/DefaultTimeSlotService.java:121]: Visio c'est du français, utilise VideoCall ou VideoConference plutot

Comment (alicedraillard) [reply backend/src/main/java/fr/openent/appointments/service/impl/DefaultTimeSlotService.java:121]: @niko-amosse valable pr tout le code

File: frontend/src/core/enums.ts [hunk 6]
```diff
@@ -24,3 +24,9 @@ export enum GRID_STATE {
   SUSPENDED = "SUSPENDED",
   CLOSED = "CLOSED",
 }
+
+export enum ALERT {
+  TAKE_APPOINTMENT_SUCCESS = "TAKE_APPOINTMENT_SUCCESS",
```
Comment (alicedraillard) [inline frontend/src/core/enums.ts:29]: BOOK_APPOINTMENT

**Review by qperie — APPROVED** [review]

**Pull request merged**